modules/ROOT/pages/keycloak/authenticating.adoc

```diff
@@ -51,19 +48,180 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
+
+. The callback provided to `login` will be invoked.
+****
+
+[role="secondary"]
+.iOS
+****
+. Use the `login` function to trigger the authentication flow:
++
+[source,swift]
+----
+try AgsAuth.instance.login(presentingViewController: self, onCompleted: onLoginComplete)
+
+// handle the authentication result
+func onLoginComplete(user: User?, err: Error?) {
+    // login failed
+    if let error = err {
+        // login error
+        return
+    }
+    // login succeeded
+    let currentUser = user
+}
+----
+
+. Allow your application to handle the redirect in `AppDelegate`:
++
+[source,swift]
+----
+func application(_ app: UIApplication, open url: URL, options: [UIApplicationOpenURLOptionsKey: Any] = [:]) -> Bool {
+    do {
+        return try AgsAuth.instance.resumeAuth(url: url as URL)
+    } catch AgsAuth.Errors.serviceNotConfigured {
+        print("AeroGear auth service is not configured")
+    } catch {
+        fatalError("Unexpected error: \(error).")
+    }
+    return false
+}
+----
+****
+
+[role="secondary"]
+.Cordova
+****
+Authentication will be performed on initialization as described xref:keycloak/coding.adoc[here]. To manually redirect to the login screen use:
+[code,javascript]
+----
+authService.login()
+----
+****
+
+[role="secondary"]
+.Xamarin
+****
+. Get an instance of `IAuthService`:
++
+[source,csharp]
+----
+IAuthService service = MobileCore.Instance.GetInstance<IAuthService>();
+----
+
+. Use the `Authenticate` method to start the auth flow:
++
+[source,csharp]
+----
+var authOptions = DependencyService.Get<IAuthenticateOptionsProvider>().GetOptions();
+service.Authenticate(authOptions).ContinueWith(result =>
+{
+    // load some view
+});
+----
+****
+
+== Logging out
+
+The following section describes how to perform a logout.
+
+[role="primary"]
+.Android
+****
+To logout, invoke the `logout` method. This accepts the UserPrincipal that was provided by `handleAuthResult` and has a callback to determine if the logout to the Keycloak or OpenID Connect server was successful.
+
+[source,java]
+----
+
+
+// authService already initialized.
+AuthService authService = MobileCore.getInstance().getService(AuthService.class);
+UserPrincipal currentUser = authService.currentUser();
+
+authService.logout(currentUser, new Callback<UserPrincipal>() {
+    @Override
+    public void onSuccess() {
+        // User Logged Out Successfully and Local Auth Tokens were Deleted
+    }
+
+    @Override
+    public void onError(Throwable error) {
+        // An error occurred during logout
+    }
+});
+
 ----
 
-The callback provided in `AuthService#login` will be invoked.
 ****
 
 [role="secondary"]
 .iOS
 ****
+. To logout invoke the `logout` method:
```

Comment:
I would change the order, first implement callback then call logout method. Also I think we could reword _post-logout actions_ for simply results or _check for errors_